oshi-core/src/main/java/oshi/util/FileUtil.java

```diff
@@ -119,7 +124,9 @@ public static List<String> readLines(String filename, int count, boolean reportE
             if (LOG.isDebugEnabled()) {
                 LOG.debug(READING_LOG, filename);
             }
-            try (BufferedReader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
+            CharsetDecoder decoder = StandardCharsets.UTF_8.newDecoder();
+            try (Reader isr = new InputStreamReader(Files.newInputStream(file), decoder);
+                    BufferedReader reader = new BufferedReader(isr, BUFFER_SIZE)) {
                 List<String> lines = new ArrayList<>(count);
```

Comment:
Since we're micro-optimizing, we really should handle the "read one line" case separately to avoid the List overhead. No need to handle that in this PR but something to consider!

Reply:
I agree. However, the savings from avoiding the list allocation are small (< 100 bytes) compared to the savings from reducing the buffer size (14kB).